ext/sqlite3/sqlite3.c

```diff
@@ -1277,6 +1277,61 @@ PHP_METHOD(sqlite3, enableExceptions)
 }
 /* }}} */
 
+#if SQLITE_VERSION_NUMBER >= 3006011
+/* {{{ proto bool SQLite3::backup(SQLite3 destination_db[, string source_dbname = "main"[, string destination_dbname = "main"]])
+   Backups the current database to another one. */
+PHP_METHOD(sqlite3, backup)
+{
+	php_sqlite3_db_object *source_obj;
+	php_sqlite3_db_object *destination_obj;
+	char *source_dbname = "main", *destination_dbname = "main";
+	size_t source_dbname_length, destination_dbname_length;
+	zval *source_zval = getThis();
+	zval *destination_zval;
+	zend_error_handling error_handling;
+	sqlite3_backup *dbBackup;
+	int rc; // Return code
+
+	source_obj = Z_SQLITE3_DB_P(source_zval);
```

Comment:
Shouldn't we check whether `source_obj` is initialized?

Reply:
Yes.